src/sentry/api/serializers/rest_framework/widget.py

```diff
@@ -0,0 +1,47 @@
+from __future__ import absolute_import
+
+from rest_framework import serializers
+
+from sentry.api.bases.discoversavedquery import DiscoverSavedQuerySerializer
+from sentry.api.serializers.rest_framework import JSONField, ListField, ValidationError
+from sentry.models import WidgetDisplayTypes, WidgetDataSourceTypes
+
+
+class WidgetDataSourceSerializer(serializers.Serializer):
+    name = serializers.CharField(required=True)
+    data = JSONField(required=True)
+    type = serializers.CharField(required=True)
+    order = serializers.IntegerField(required=True)
+
+    def validate_type(self, attrs, source):
+        type = attrs[source]
+        if type not in WidgetDataSourceTypes.TYPE_NAMES:
+            raise ValidationError('Widget data source type %s not recognized.' % type)
+        attrs[source] = WidgetDataSourceTypes.get_id_for_type_name(type)
+        return attrs
+
+    def validate(self, data):
+        if data['type'] == WidgetDataSourceTypes.DISCOVER_SAVED_SEARCH:
+            serializer = DiscoverSavedQuerySerializer(data=data['data'], context=self.context)
+            if not serializer.is_valid():
+                raise ValidationError('Error validating DiscoverSavedQuery: %s' % serializer.errors)
+        return data
+
+
+class WidgetSerializer(serializers.Serializer):
+    displayType = serializers.CharField(required=True)
+    displayOptions = JSONField(required=False)
+    title = serializers.CharField(required=True)
+    dataSources = ListField(
+        child=WidgetDataSourceSerializer(required=False),
+        required=False,
+        allow_null=True,
+    )
+
+    def validate_displayType(self, attrs, source):
+        display_type = attrs[source]
+        if display_type not in WidgetDisplayTypes.TYPE_NAMES:
+            raise ValidationError('Widget display_type %s not recognized.' % display_type)
```

Comment:
```suggestion
            raise ValidationError('Widget displayType %s not recognized.' % display_type)
```

Error message casing should match the field the user sent us.